utils/ixdtf/src/error.rs

```diff
@@ -12,6 +12,8 @@ use displaydoc::Display;
 pub enum ParseError {
     #[displaydoc("Implementation error: this error must not throw.")]
     ImplAssert,
+    #[displaydoc("UTF16 digit was not mappable to an ASCII value")]
+    Utf16NonAsciiChar,
```

Comment:
Nit: There isn't anything about this which is UTF-16 specific. You could make the error `NonAscii` and return it when either UTF-16 or UTF-8 contains a non-ASCII char.